### PR TITLE
.gitignore: add autogenerated files

### PR DESCRIPTION
**Pull request by black-desk**
After a build:
git show Untracked files below
  src/include/config.h
  src/include/config.h.in
  src/include/stamp-h1
  src/include/version.h

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,8 +13,8 @@ Makefile.in
 /ar-lib
 /autom4te.cache
 /compile
-/config.h
-/config.h.in
+*config.h
+*config.h.in
```
Comment (fingolfin) [inline .gitignore:17]:
```suggestion
config.h
config.h.in
```

File: .gitignore [hunk 2]
```diff
@@ -23,8 +23,8 @@ Makefile.in
 /install-sh
 /missing
 /mosh-*.tar.gz
-/stamp-h1
+*stamp-h1
```
Comment (fingolfin) [inline .gitignore:26]:
```suggestion
stamp-h1
```

File: .gitignore [hunk 3]
```diff
@@ -23,8 +23,8 @@ Makefile.in
...
 /test-driver
 /VERSION
 /scripts/mosh
-/version.h
+*version.h
```
Comment (fingolfin) [inline .gitignore:30]:
```suggestion
version.h
```

**Review by fingolfin — COMMENTED** [review]
With your changes, it will also ignore files named `fooconfig.h` etc. -- if you just drop the leading slash, it will ignore `config.h` in any directory, which I think is what you intended?

**Comment by black-desk** [comment]
maybe only ignore the files in src/include is better
this time I ignore complie_commands.json too

**Review by eminence — APPROVED** [review]

**Comment by eminence** [comment]
LGTM, thanks

**Pull request merged**